core/domain/user_services.py

```diff
@@ -2171,3 +2173,22 @@ def create_login_url(return_url):
     # flow on top of the Firebase SDK in "core/templates/pages/login-page", and
     # this function will always redirect to its static location ("/login").
     return '/login?%s' % python_utils.url_encode({'return_url': return_url})
+
+
+def update_roles_and_banned_fileds(user_settings_model):
+    """Updates the new roles and banned fileds in the UserSettingsModel which
+    are not in use but needs to kept in sync.
+
+    This function is expected to be remove soon once the roles and banned fields
+    will be in use. It is not recommended to use this function in new places.
+
+    Args:
+        user_settings_model: UserSettingsModel. The models which needs update.
+    """
+    if user_settings_model.role not in [
+            feconf.ROLE_ID_LEARNER, feconf.ROLE_ID_EXPLORATION_EDITOR]:
+        user_settings_model.roles = [user_settings_model.role]
```

Comment:
Hmm, looking at the design doc, should we fill this with all the roles below the current roles? Except for the learners and exp. editors.

Reply:
The design is changed now and I've updated the job accordingly! (Done!)